### PR TITLE
use loki

### PR DESCRIPTION
**Pull request by podviaznikov**
Use loki for Docker/Swarm interactions
- [x] @anandkumarpatel 

Test on epsilon:
- [x] @anandkumarpatel

### CODE REVIEW

File: configs/.env.test [hunk 1]
```diff
@@ -1,5 +1,5 @@
 DOCKER_CERT_PATH=./test/fixtures
-LOG_LEVEL_STDOUT=fatal
+LOG_LEVEL_STDOUT=trace
```
Comment (anandkumarpatel) [inline configs/.env.test]: really?

File: lib/docker.js [hunk 2]
```diff
@@ -72,19 +25,7 @@ module.exports = class Docker {
...
-   * gets inspect data from container
-   * @return {Promise}
-   * @resolves {Object} inspect data
-   */
-  inspectContainer (id) {
-    log.info({ id: id }, 'Docker.prototype.inspectContainer')
-    return Promise.fromCallback((cb) => {
-      this.docker.getContainer(id).inspect(cb)
-    })
+    return this.client.getEventsAsync(opts)
```
Comment (anandkumarpatel) [inline lib/docker.js]: since we are doing inheritance here, we do not need to have this function ?

File: lib/docker.js [hunk 3]
```diff
@@ -96,7 +37,7 @@ module.exports = class Docker {
   testEvent () {
     log.info('Docker.prototype.testEvent')
     const listTimer = datadog.timer('listContainers.time')
-    return this.docker.listContainersAsync({
+    return this.client.listContainersAsync({
```
Comment (anandkumarpatel) [inline lib/docker.js]: hmm looking at `loki` not sure why you do not want `listContainersAsync` on the object itself, why put it on `client`?
ex. `this.listContainersAsync` should work

File: lib/swarm.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  getEvents (opts) {
+    log.info({ opts: opts }, 'Swarm.prototype.getEvents')
+    return this.client.getEventsAsync(opts)
+  }
+
+  /**
+   * Do docker top on a random container.
+   * This should emit the `top` event.
+   * We ignore errors since we will restart if event failed
+   * @param  {Function} cb [description]
```
Comment (anandkumarpatel) [inline lib/swarm.js]: description

File: lib/swarm.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      listTimer.stop()
+      if (isEmpty(containers)) {
+        throw new Error('no running containers found')
+      }
+      var testContainerId = containers[0].Id
+      log.trace({ container: testContainerId }, 'testEvent - got container')
+
+      const topTimer = datadog.timer('top.time')
+      return this.topContainer(testContainerId)
+      .then(() => {
```
Comment (anandkumarpatel) [inline lib/swarm.js]: nit: space?

File: configs/.env [hunk 6]
```diff
@@ -1,3 +1,4 @@
+APP_NAME=docker.listener
```
Comment (anandkumarpatel) [inline configs/.env]: I think we should split this up into sections. If I grep for `APP_NAME` in this repo, I would not get anything and think i should delete this.

I would add a section below with comment

```
# configs for loki
APP_NAME=docker.listener
```

File: lib/docker.js (hunk 2 above)
Comment (podviaznikov) [inline lib/docker.js]: yes, removed

File: lib/docker.js (hunk 3 above)
Comment (podviaznikov) [inline lib/docker.js]: updated, agreed

File: configs/.env (hunk 6 above)
Comment (podviaznikov) [inline configs/.env]: addressed this one

File: .travis.yml [hunk 7]
```diff
@@ -1,6 +1,9 @@
 language: node_js
 node_js:
 - '4.2'
+- '4'
+- '5'
+- '6'
```
Comment (anandkumarpatel) [inline .travis.yml:6]: do we need to test all these?

Comment (podviaznikov) [inline .travis.yml:6]: I think it's good to have them for future. If something will fail we wouldn't know that we can't migrate to newest node

File: lib/swarm.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      })
+  }
+
+  /**
+   * Do docker top on a random container.
+   * This should emit the `top` event.
+   * We ignore errors since we will restart if event failed
+   * @param  {Function} cb
+   */
+  testEvent () {
```
Comment (anandkumarpatel) [inline lib/swarm.js]: I dont like that this logic is duplicated ... maybe add this to the base class? or make a base class that does this.

Comment (anandkumarpatel) [inline lib/swarm.js]: Or abstract it out somewhere, this is a lot of logic to have copied in 2 places

File: lib/docker.utils.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+const datadog = require('./datadog')
+const isEmpty = require('101/is-empty')
+const log = require('./logger')()
+
+/**
+ * Do docker top on a random container.
+ * This should emit the `top` event.
+ * We ignore errors since we will restart if event failed
+ * @param  {Function} cb
```
Comment (anandkumarpatel) [inline lib/docker.utils.js]: @returns promise
@resolves (null) when test complete
@rejects (err) if test failed or timeout

File: lib/docker.utils.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
+'use strict'
```
Comment (anandkumarpatel) [inline lib/docker.utils.js]: nit: file names are normally split with `-` not `.`

**Comment by anandkumarpatel** [comment]
LGTM

**Comment by podviaznikov** [comment]
demoed for @anandkumarpatel. Also tested dock unhealthy and dock kill

**Pull request merged**